tests/test_mux.py

```diff
@@ -664,6 +665,45 @@ def remove_and_test_tunnel(self, db, asicdb, tunnel_name):
         assert not self.check_interface_exists_in_asicdb(asicdb, overlay_infs_id)
 
 
+    def remove_link_and_test_tunnel_create(self, appdb, asicdb, confdb, dvs):
+        self.create_vlan_interface(confdb, asicdb, dvs)
+        self.create_mux_cable(confdb)
+        self.create_and_test_tunnel(appdb, asicdb, tunnel_name="MuxTunnel0", tunnel_type="IPINIP",
+                                   dst_ip="10.1.0.32", dscp_mode="uniform",
+                                   ecn_mode="standard", ttl_mode="pipe")
+        
+        peer_attrs = {
```

Comment:
Please add the following tests as well:
1. Add a FAILED neighbor before configuring peer_switch. There should not be a tunnel route
2. Do mux status changes after resolving the entry and verify neighbor, and tunnel route
3. Add a FAILED entry and delete the entry. Make sure the tunnel route is removed as well.